src/client.cpp

```diff
@@ -473,7 +473,7 @@ void UpdatePort(unsigned int port, bool connected, const game_controller* contro
 
   const unsigned int device = CInputManager::Get().GetDevice(port);
 
-  if (CLIENT)
+  if (CLIENT && (int)port >= 0)
```

Comment:
Negative ports should be an implementation detail of the game API. Ports everywhere else should be positive, and if necessary declare that ports are specifically joystick/controller ports.